### PR TITLE
Implemented some bytecodes

### PR DESCRIPTION
**Pull request by tsarn**
Also fixed int storage semantics

### CODE REVIEW

File: src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp [hunk 1]
```diff
@@ -541,31 +556,24 @@ void InterpreterMacroAssembler::index_check_without_pop(Register Rarray, Registe
...
   // Index check
   lwu(Rlength, Rarray, arrayOopDesc::length_offset_in_bytes());
-
-  blt(RsxtIndex, Rlength, LnotOOR);
-  // Index should be in R25_tos, array should be in R4_ARG2_PPC.
+  blt(Rindex, Rlength, LnotOOR);
+  // Index should be in R25_tos, array should be in R12_ARG2.
   mv_if_needed(R25_tos, Rindex);
-  mv_if_needed(R11_ARG1, Rarray);
+  mv_if_needed(R12_ARG2, Rarray);
```
Comment (Geckostya) [inline src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp]: PPC starts args from 1, RISCV from 0, hence PPC_ARG2 should be passed into RISCV_ARG1

File: src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp [hunk 2]
```diff
@@ -541,31 +556,24 @@ void InterpreterMacroAssembler::index_check_without_pop(Register Rarray, Registe
     //null_check_throw(Rarray, arrayOopDesc::length_offset_in_bytes(), /*temp*/RsxtIndex); FIXME_RISCV
   }
 
-  // Rindex might contain garbage in upper bits (remember that we don't sign extend
-  // during integer arithmetic operations). So kill them and put value into same register
-  // where ArrayIndexOutOfBounds would expect the index in.
-  slli(RsxtIndex, Rindex, 32);
```
Comment (Geckostya) [inline src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp:547]: why you have removed these operations?

File: src/hotspot/cpu/riscv/templateTable_riscv_64.cpp [hunk 3]
```diff
@@ -121,21 +118,25 @@ void TemplateTable::patch_bytecode(Bytecodes::Code new_bc, Register Rnew_bc, Reg
...
+      __ lbu(Rnew_bc, Rtemp, in_bytes(ConstantPoolCache::base_offset() + ConstantPoolCacheEntry::indices_offset()) + 7 - (1 + byte_no));
 #endif
-      __ cmpwi_PPC(CCR0, Rnew_bc, 0);
-      __ li_PPC(Rnew_bc, (unsigned int)(unsigned char)new_bc);
-      __ beq_PPC(CCR0, L_patch_done);
-      // __ isync_PPC(); // acquire not needed
+      __ beqz(Rnew_bc, L_zero);
+      __ li(Rnew_bc, (unsigned int)(unsigned char)new_bc);
+      __ sb(Rnew_bc, R22_bcp, 0);
+      __ j(L_patch_done);
```
Comment (Geckostya) [inline src/hotspot/cpu/riscv/templateTable_riscv_64.cpp]: In the original code you jump to the done label only if Rnew_bc is zero. Now you do this in each case

File: src/hotspot/cpu/riscv/templateTable_riscv_64.cpp [hunk 4]
```diff
@@ -171,20 +172,20 @@ void TemplateTable::shouldnotreachhere() {
...
   assert(value >= -1 && value <= 5, "");
-  __ addi(R25_tos, R0, value);
+  __ addi(R25_tos, R0_ZERO, value);
 }
 
 void TemplateTable::lconst(int value) {
   transition(vtos, ltos);
   assert(value >= -1 && value <= 5, "");
-  __ li_PPC(R25_tos, value);
+  __ addi(R25_tos, R0_ZERO, value);
```
Comment (Geckostya) [inline src/hotspot/cpu/riscv/templateTable_riscv_64.cpp]: li is prefer

File: src/hotspot/cpu/riscv/templateTable_riscv_64.cpp [hunk 5]
```diff
@@ -1494,82 +1503,80 @@ void TemplateTable::convert() {
   Label done;
   switch (bytecode()) {
```
Comment (Geckostya) [inline src/hotspot/cpu/riscv/templateTable_riscv_64.cpp:1505]: Hard code. @vasyoid @DmKazakov please check it, you are more familiar with such PPC instructions

File: src/hotspot/cpu/riscv/templateTable_riscv_64.cpp [hunk 6]
```diff
@@ -2192,10 +2165,7 @@ void TemplateTable::_return(TosState state) {
   __ remove_activation(state, /* throw_monitor_exception */ true);
   // Restoration of lr done by remove_activation.
   switch (state) {
-    // Narrow result if state is itos but result type is smaller.
-    // Need to narrow in the return bytecode rather than in generate_return_entry
-    // since compiled code callers expect the result to already be narrowed.
-    case itos: __ narrow(R25_tos); /* fall through */
+    case itos:
```
Comment (Geckostya) [inline src/hotspot/cpu/riscv/templateTable_riscv_64.cpp:2169]: why did you remove the narrowing?

**Review by Geckostya — CHANGES_REQUESTED** [review]
I did not see it completely, because I did not understand everything well

File: src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp (hunk 2 above)
Comment (tsarn) [reply src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp:547]: registers are always correctly sign-extended on riscv

File: src/hotspot/cpu/riscv/templateTable_riscv_64.cpp (hunk 3 above)
Comment (tsarn) [reply src/hotspot/cpu/riscv/templateTable_riscv_64.cpp]: Works as intended

File: src/hotspot/cpu/riscv/templateTable_riscv_64.cpp (hunk 6 above)
Comment (tsarn) [reply src/hotspot/cpu/riscv/templateTable_riscv_64.cpp:2169]: registers are always correctly sign-extended on riscv

File: src/hotspot/cpu/riscv/templateTable_riscv_64.cpp (hunk 3 above)
Comment (Geckostya) [reply src/hotspot/cpu/riscv/templateTable_riscv_64.cpp]: You skip code in `if (JvmtiExport::can_post_breakpoint()) {`
